xclim/core/calendar.py

```diff
@@ -1588,3 +1589,302 @@ def get_doys(start, end, inclusive):
         mask["time"] = da.time
 
     return da.where(mask, drop=drop)
+
+
+def _month_is_first_period_month(time, freq):
+    """Returns True if the given time is from the first month of freq."""
+    if isinstance(time, cftime.datetime):
+        frqM = xr.coding.cftime_offsets.to_offset("MS")
+        frq = xr.coding.cftime_offsets.to_offset(freq)
+        if frqM.onOffset(time):
+            return frq.onOffset(time)
+        return frq.onOffset(frqM.rollback(time))
+    # Pandas
+    time = pd.Timestamp(time)
+    frqM = pd.tseries.frequencies.to_offset("MS")
+    frq = pd.tseries.frequencies.to_offset(freq)
+    if frqM.is_on_offset(time):
+        return frq.is_on_offset(time)
+    return frq.is_on_offset(frqM.rollback(time))
```

Comment:
Could we convert to a cftime and just recall the function? I tried and obtaining a cftime is not super obvious with the APIs?

Reply:
e.g. 
```suggestion
    # Pandas
    time = pd.Timestamp(time)
    time = cftime.datetime(*[getattr(time, meth) for meth in ["year", "month", "day", "hour", "minute", "second"]])
    _month_is_first_period_month(time, freq)
```

maybe it's more complicated or could lead to problem, feel free to ignore

Reply:
You can leave it as it is then, will try to think about `pd.Timestamp`